CRM/Contact/BAO/Query.php

```diff
@@ -3173,21 +3176,22 @@ public function tagSearch(&$values) {
 
     $op = "LIKE";
     $value = "%{$value}%";
+    $escapedValue = CRM_Utils_Type::escape("%{$value}%", 'String');
 
     $useAllTagTypes = $this->getWhereValues('all_tag_types', $grouping);
     $tagTypesText = $this->getWhereValues('tag_types_text', $grouping);
 
-    $etTable = "`civicrm_entity_tag-" . $value . "`";
-    $tTable = "`civicrm_tag-" . $value . "`";
+    $etTable = "`civicrm_entity_tag-" . uniqid() . "`";
```

Comment:
@seamuslee001 maybe the group one needs to look more like this?

Reply:
@eileenmcnaughton that is exactly my thinking :-)

Reply:
were you also thinking the other thing I'm thinking?

Reply:
I'm not sure B1

Reply:
:-)